common/rsyslog_plugin.cpp

```diff
@@ -0,0 +1,161 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <cstdlib>
+#include <cstring>
+#include <fstream>
+#include <unordered_set>
+#include <regex>
+#include <tuple>
+#include "logger.h"
+#include "json.hpp"
+
+using namespace std;
+using namespace nlohmann;
+
+string parserCfgFile = "{}_parse.rc.json";
+ofstream outfile;
+string logfile = "/tmp/{}_logfile";
+json regexList = json::array();
+string pgmName = "";
+unordered_set<string> errReportedTags;
+
+void loadRegexlist(string rcPath, string pgmName) {
+	fstream rcfile;
+	parserCfgFile = regex_replace(parserCfgFile, regex("\\{}"), pgmName);
+	parserCfgFile = rcPath + parserCfgFile;
+	rcfile.open(parserCfgFile, ios::in);
+	if (!rcfile) {
+		cerr << "No such file: " << parserCfgFile << endl;
+		return;
+	}
+	rcfile >> regexList;
```

Comment:
>rcfile [](http://example.com/codeflow?start=1&length=6)

Without a design doc or unit test, it's hard to understand the input file schema.